api/envoy/extensions/filters/http/ext_proc/v3/processing_mode.proto

```diff
@@ -68,6 +69,31 @@ message ProcessingMode {
     // chunk. If the body exceeds the configured buffer limit, then the body contents
     // up to the buffer limit will be sent.
     BUFFERED_PARTIAL = 3;
+
+    // [#not-implemented-hide:]
+    // Envoy streams the body to the server in pieces as they arrive. The data is not stored in
+    // the internal queue. Only data size is stored there for tracking and flow control purpose.
+    //
+    // Server may choose to buffer any number, i.e M, chunks of data before processing them.
+    // After finish buffering, server processes the buffered data. Then it splits the processed
+    // data into any number, i.e N, chunks, and stream them back to Envoy one by one.
+    // The server may continuously do so until the complete body is processed.
+    //
+    // Server may also choose to buffer the entire message, including the headers(if header mode is
+    // ``SEND``), the entire body, and the trailers(if present), before sending back any response.
+    // The server response has to maintain the headers-body-trailers ordering.
+    //
+    // In this body mode:
+    // * The corresponding trailer mode has to be set to ``SEND``.
+    // * Envoy will send body and trailers(if present) as they arrive.
+    //   Sending the trailers(if present) is to inform the server the complete body arrives.
+    //   In case of no trailers, then
+    //   :ref:`end_of_stream <envoy_v3_api_field_service.ext_proc.v3.HttpBody.end_of_stream>`
+    //   serves for that purpose.
+    // * Server needs to send
+    //   :ref:`MxnBodyResponse <envoy_v3_api_msg_service.ext_proc.v3.MxnBodyResponse>`
+    //   to Envoy in the body response.
+    MXN = 4;
```

Comment:
I haven't review the full PR but just want to raise the discussion around naming here earlier. Control/management plane folks are also waiting for the naming here.

I personally feel we should give a more descriptive name. `MXN` sounds like a internal code name.

Maybe we can call it something like parallel_streaming?  Open to other opinions.  @yanavlasov  @rshriram @htuch

Reply:
How about BIDIRECTIONAL_STREAMING ?

Reply:
IMHO, `BIDIRECTIONAL_STREAMING` could be confused with gRPC bi-directional streaming mode which is already used in all ext_proc processing mode.

Reply:
A simple rule of thumb is that variable names should be self-explanatory.

If you look through the code base or just look at the other processing mode names here, they at least give some sense of the nature of the modes.

what is MXN ?  `M`,  `X`,  `N` ? M times N? I don't think this name provides any useful information, especially to people who is new to this feature.

Reply:
In that case BIDIRECTIONAL_STREAMING seems to be better. I don't feel people will confuse it with gRPC bi-directional streaming.  Same as "STREAMED" mode which is widely used, but here people will interpret it under ext_proc context.

Reply:
My take is not using `BIDIRECTIONAL_STREAMING` that is confusing with gRPC widely used term. I think there are so many options out there can describe this mode.

@mkosieradzki and I just discussed our internal api and he has suggested FULL_DUPLEX_STREAMED.

Reply:
@yanavlasov @adisuissa could you please chime in on this API naming?

Also, how about continue reviewing the changes in the state machine while we are waiting for comments from other stakeholders?

Reply:
drive-by comment: some other options could be { dynamic | flexible | non-blocking | on_demand | sidestream_driven }_streaming. Whatever you choose, please consistently replace references to MxN throughout your change.

Reply:
FULL_DUPLEX_STREAMED

And replaced the references to by this name across the PR.